addons/onboarding/models/onboarding_onboarding.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class Onboarding(models.Model):
+    _name = 'onboarding.onboarding'
+    _description = 'Onboarding'
+
+    name = fields.Char('Name of the onboarding', translate=True)
+    route_name = fields.Char('One word name', help='One word identifier used for technical purposes.')
+    step_ids = fields.One2many('onboarding.onboarding.step', 'onboarding_id', 'Onboarding steps')
+
+    is_per_company = fields.Boolean('Should be done per company?', default=True)
+
+    panel_background_color = fields.Selection(
+        [('orange', 'Orange'), ('blue', 'Blue'), ('violet', 'Violet'), ('none', 'None')],
+        string="Panel's Background color", default='orange',
+        help="Color gradient added to the panel's background.")
+    panel_background_image = fields.Binary("Panel's background image")
+    panel_close_action_name = fields.Char(
+        'Closing action', help='Name of the onboarding model action to execute when closing the panel.')
+
+    current_progress_id = fields.Many2one(
```

Comment:
@tde-banana-odoo  I didn't manage to fetch the right `onboarding_progress_step` from the `onboarding.current_progress_id` (the `onboarding.progress` record don't have an easy way to identify which of the `progress_step_ids` is for which step).

I tried to set this one to `one2Many` (required because `progress.progress_step_ids` is X2Many too) with `related='onboarding_id.current_progress_id.progress_step_ids'` with a `domain="[('step_id' '=' 'id')]"` but the domain wasn't applied and I didn't get the progress.step of the right step.

Is it possible then? If so, what did I miss?